ui/apps/platform/src/Containers/Policies/Wizard/Step2/PolicyBehaviorForm.tsx

```diff
@@ -19,25 +19,47 @@ import {
 } from '@patternfly/react-core';
 import { useFormikContext } from 'formik';
 
+import ConfirmationModal from 'Components/PatternFly/ConfirmationModal';
 import { ClientPolicy, LifecycleStage } from 'types/policy.proto';
 
 import {
     appendEnforcementActionsForAddedLifecycleStage,
     filterEnforcementActionsForRemovedLifecycleStage,
     hasEnforcementActionForLifecycleStage,
+    initialPolicy,
 } from '../../policies.utils';
 import DownloadCLIDropdown from './DownloadCLIDropdown';
 
 import './PolicyBehaviorForm.css';
 
-function PolicyBehaviorForm() {
+type PolicyBehaviorFormProps = {
+    hasActiveViolations: boolean;
+};
+
+function PolicyBehaviorForm({ hasActiveViolations }: PolicyBehaviorFormProps) {
     const { values, setFieldValue, setValues } = useFormikContext<ClientPolicy>();
+    const [lifeCycleChange, setLifeCycleChange] = useState<{
+        lifecycleStage: LifecycleStage;
+        isChecked: boolean;
+    } | null>(null);
+
     const hasEnforcementActions =
         values.enforcementActions?.length > 0 &&
         !values.enforcementActions?.includes('UNSET_ENFORCEMENT');
     const [showEnforcement, setShowEnforcement] = React.useState(hasEnforcementActions);
 
     function onChangeLifecycleStage(lifecycleStage: LifecycleStage, isChecked: boolean) {
+        if (values?.id) {
+            // for existing policies, warn that changing lifecycles will clear all policy criteria
+            setLifeCycleChange({ lifecycleStage, isChecked });
+        } else {
+            // for new policies, just update lifecycle stages
+            const newValues = getLifeCyclesUpdates(lifecycleStage, isChecked);
+            setValues(newValues);
+        }
+    }
+
+    function getLifeCyclesUpdates(lifecycleStage: LifecycleStage, isChecked: boolean) {
```

Comment:
This is good. In the spirit of caution like following comment, what do you think about:
1. Extract the function to utils to ensure no unintended references to function scope.
2. Therefore make `values` the explicit first argument of the function
3. Replace `...values` with `cloneDeep(values)` because object spread only guarantees pure update at the first level, not deeper levels.